cg/meta/workflow/mutant/metrics_parser/metrics_parser.py

```diff
@@ -0,0 +1,131 @@
+    VOC: str = "VOC"
+    MUTATIONS: str = "Mutations"
+
+
+class MetricsParser:
+    KEY_MAPPING: dict[str, str] = {
+        MutantResultsHeaderRawData.SAMPLE_NAME: MutantResultsHeaderData.SAMPLE_NAME,
+        MutantResultsHeaderRawData.SELECTION: MutantResultsHeaderData.SELECTION,
+        MutantResultsHeaderRawData.REGION_CODE: MutantResultsHeaderData.REGION_CODE,
+        MutantResultsHeaderRawData.TICKET: MutantResultsHeaderData.TICKET,
+        MutantResultsHeaderRawData.PCT_N_BASES: MutantResultsHeaderData.PCT_N_BASES,
+        MutantResultsHeaderRawData.PCT_10X_COVERAGE: MutantResultsHeaderData.PCT_10X_COVERAGE,
+        MutantResultsHeaderRawData.QC_PASS: MutantResultsHeaderData.QC_PASS,
+        MutantResultsHeaderRawData.LINEAGE: MutantResultsHeaderData.LINEAGE,
+        MutantResultsHeaderRawData.PANGOLIN_DATA_VERSION: MutantResultsHeaderData.PANGOLIN_DATA_VERSION,
+        MutantResultsHeaderRawData.VOC: MutantResultsHeaderData.VOC,
+        MutantResultsHeaderRawData.MUTATIONS: MutantResultsHeaderData.MUTATIONS,
+    }
+
+    @classmethod
+    def parse_samples_results(cls, case: Case, results_file_path: Path) -> dict[str, SampleResults]:
+        """Takes a case object and a results_file_path and resturns dict[str, SampleResults] with sample.internal_id as keys."""
+
+        raw_results: list[dict[str, Any]] = cls._get_raw_results(
+            results_file_path=results_file_path
+        )
+
+        validated_results_list: list[SampleResults] = cls._get_validated_results_list(
+            raw_results=raw_results
+        )
+
+        samples_results: dict[str, SampleResults] = cls._get_samples_results(
+            case=case, results_list=validated_results_list
+        )
+
+        return samples_results
+
+    @classmethod
+    def _get_raw_results(cls, results_file_path: Path) -> list[dict[str, Any]]:
+        """Parses raw_results from the results file."""
+        try:
+            raw_results: list[dict[Any, Any]] = [
+                sample_results
+                for sample_results in read_csv(file_path=results_file_path, read_to_dict=True)
+            ]
+        except FileNotFoundError as exception_object:
+            raise CgError(f"Results file not found at {results_file_path}") from exception_object
+        except Exception as exception_object:
+            raise CgError(
+                f"Not possible to read results file {results_file_path}."
+            ) from exception_object
+        return raw_results
+
+    @classmethod
+    def _get_altered_sample_result(cls, sample_result: dict[str, Any]) -> dict[str, Any]:
+        """Takes a raw_sample_result with headers from the results file (MutantResultsHeaderRawData)
+        and returns an altered_sample_result with the corrected headers from MutantResultsHeaderData.
+        """
+        altered_sample_result = {}
+        for header, value in sample_result.items():
+            new_header: str = cls.KEY_MAPPING.get(header, header)
```

Comment:
Why do we use use get with "header" twice?

Reply:
From the `KEY_MAPPING` dict I want to retrieve the value for that `header` key, and if the key is not found in the dict, the value should stay the same aka `header` itself.

`dictionary.get(keyname, value)`

Parameter | Description
-- | --
keyname | Required. The keyname of the item you want to return the value from
value | Optional. A value to return if the specified key does not exist.Default value None

https://www.w3schools.com/python/ref_dictionary_get.asp

Reply:
ah, it was a default value, fair enough. So it is not a problem if the key is not found?

Reply:
It will just fail validation later! 🤷‍♀️